src/parser.rs

```diff
@@ -3062,10 +3011,11 @@ fn parse_type_unit_header<Endian>(input: EndianBuf<Endian>)
                                   -> ParseResult<(EndianBuf<Endian>, TypeUnitHeader<Endian>)>
     where Endian: Endianity
 {
-    let (rest, header) = try!(parse_unit_header(input));
-    let (rest, signature) = try!(parse_type_signature(rest));
+    let (after_unit, mut header) = try!(parse_unit_header(input));
+    let (rest, signature) = try!(parse_type_signature(header.entries_buf));
     let (rest, offset) = try!(parse_type_offset(rest, header.format()));
-    Ok((rest, TypeUnitHeader::new(header, signature, offset)))
+    header.entries_buf = rest;
+    Ok((after_unit, TypeUnitHeader::new(header, signature, offset)))
```

Comment:
This is way nicer now with the split between the stuff after the unit and the rest of the unit. Thanks!